sema4ai/src/sema4ai_code/refresh_agent_spec_helper.py

```diff
@@ -0,0 +1,128 @@
+from pathlib import Path
+from typing import Optional, TypedDict
+
+from ruamel.yaml import YAML
+
+
+class ActionPackage(TypedDict, total=False):
+    name: str
+    organization: str
+    type: str
+    version: str
+    whitelist: str
+    path: str
+
+
+class WhitelistEntry(TypedDict):
+    name: str
+    version: Optional[str]
+    organization: str
+    path: str
+    whitelist: Optional[str]
+
+
+def load_yaml(filepath: Path, yaml: YAML) -> None:
+    with filepath.open("r") as file:
+        return yaml.load(file)
+
+
+def save_yaml(data: dict, filepath: Path, yaml: YAML) -> None:
```

Comment:
I'd just inline this function as it's only used once and it's simple enough to not need a new function.